File: Problem 2/problem2.java
/**
 * Project Euler Problem 2
 * @author Tyler Smith
 * @email tsmith328@gatech.edu
 */

import java.util.List;
import java.util.ArrayList;

class problem2 {
    public static void main(String[] args) {
        List<Integer> fibs = getFibs(4000000);
        fibs = onlyEven(fibs);
        int sum = 0;
        for (int i : fibs) {
            sum += i;
        }
        System.out.println("The sum of all even Fibonacci numbers less than 4 million is:");
        System.out.println(sum);
    }

    static List<Integer> getFibs(int limit) {
        int a = 1;
        int b = 1;
        List<Integer> fibs = new ArrayList<Integer>();
        while (b < limit) {
            fibs.add(b);
            int t = a + b;
            a = b;
            b = t;
        }
        return fibs;
    }

    static List<Integer> onlyEven(List<Integer> list) {
        List<Integer> newList = new ArrayList<Integer>();
        for (int i : list) {
            if (i % 2 == 0) {
                newList.add(i);
            }
        }
        return newList;
    }
}

File: Problem 2/problem2.py
# Project Euler
# Tyler M Smith
# tsmith328@gatech.edu

def main():
    fibs = getFibs(4000000)
    fibs = onlyEven(fibs)
    print("The summ of all Fibonacci numbers less than 4 million is:")
    print(sum(fibs))

def onlyEven(aList):
    newList = [x for x in aList if x % 2 == 0]
    return newList

def getFibs(maxVal):
    a = 1
    b = 2
    retList = []
    retList.append(a)
    retList.append(b)
    newVal = a + b
    while newVal < 4000000:
        retList.append(newVal)
        a = b
        b = newVal
        newVal = a + b
    return retList

if __name__ == "__main__":
    main()